pyccel/ast/variable.py

```diff
@@ -521,6 +526,10 @@ class ValuedVariable(Variable):
     >>> n
     n := 4
     """
+    #TODO: Can this have children?
+    # We have a problem when we do `dtype=float` as `float`
+    # becomes a class, not a Pyccel object
+    #_attribute_nodes = ('_value',)
```

Comment:
This should be discussed before merging

Reply:
A `ValuedVariable`  is used for default arguments of a function. E.g.:
`def f(a = 3):`
the `_value`  is `3`  which should be in the `_attribute_nodes` , but when we do:
`x = empty(dtype=float)`
`dtype=float`  is a `ValuedVariable`  and `float`  is a type not a `Basic`  so it shouldn't be in `_attribute_nodes`

Reply:
@yguclu 

As I had to add exceptions in my code for `PyccelSymbol` which is immutable and therefore cannot inherit from `Basic` I have added the type `type` to that list of exceptions. This fixes the issue described above.
This is done in these 2 commits: 5b32617  1e1ec11